methods/get_seqrun_meta.nf

```diff
@@ -1,9 +1,21 @@
 // Function for sequencing_run & clarity_sample_id
 def get_seqrun_meta(LinkedHashMap row) {
-    if (row.sequencing_run && row.clarity_sample_id) {
-        meta = tuple(row.id, row.sequencing_run, row.clarity_sample_id)
+    if (row.sequencing_run) {
+        meta = [row.id, row.sequencing_run]
+        if (row.clarity_sample_id) {
+            meta << row.clarity_sample_id
+        } else {
+            row.clarity_sample_id = row.id
+            meta << row.clarity_sample_id
+        }
+        if (row.sample_name) {
+            meta << row.sample_name
+        } else {
+            row.sample_name = row.id
+            meta << row.sample_name
+        }
```

Comment:
I'd suggest simplifying this to something like the following, that should be equivalent but simpler:

```groovy
if (!row.clarity_sample_id) {
    row.clarity_sample_id = row.id
}
meta << row.clarity_sample_id
          
if (!row.sample_name) {
    row.sample_name = row.id
}
meta << row.sample_name
```

Reply:
The whole point is to make the `clarity_sample_id` and `sample_name` optional. Your suggestion assumes them to be in the csv.

Reply:
Hmm, the new example should be identical semantically, since

```groovy
meta << row.clarity_sample_id
```

... as well as:

```groovy
meta << row.sample_name
```

... are currently duplicated in both the `if` and the `else` branches, meaning they will be added either way.

So, in the current code, the only thing that differs between the `if` and `else` part is setting the values from `row.id` if they don't exist, which is better to do with a single if-statement, as branched if/else branching is known to be a common source of bugs in general, and good to avoid if possible.

Reply:
Oh yes I see what you mean now